src/ui.rs

```diff
@@ -26,13 +26,33 @@ fn calculate_geometry(window: &Node, output: &Node, args: Arc<Args>) -> (i32, i3
     (rel_x - anchor_x, rel_y - anchor_y)
 }
 
-fn handle_keypress(conn: Arc<Mutex<Connection>>, key_to_con_id: &HashMap<char, i64>, keyval: &str) {
+fn handle_keypress(
+    conn: Arc<Mutex<Connection>>,
+    key_to_con_id: &HashMap<char, i64>,
+    keyval: &str,
+    command: &Commands,
+) {
     if keyval.len() == 1 {
         // we can unwrap because the keyval has one character
         let c = keyval.chars().next().unwrap();
         if c.is_alphabetic() && c.is_lowercase() {
             let con_id = key_to_con_id[&c];
-            sway::focus(conn, con_id);
+
+            match &command {
+                Commands::Focus => {
+                    sway::focus(&conn, con_id);
+                }
+                Commands::Swap { focus } => {
+                    sway::swap(&conn, con_id);
+
+                    if *focus {
```

Comment:
I can't remember if there's a way to use `ref` to extract `focus` from the pattern, which would allow us to avoid the explicit dereference here; I think the code would look cleaner that way. What are your thoughts?

Reply:
I was not able to remove the dereference.  I tried the `ref` keyword and a few other things but they don't seem to have any effect.